### ISSUE TITLE
Error compiling on Cygwin

### ISSUE THREAD

**Issue by crd477** [janestreet/core_kernel#49]
When I attempt to install from opam, I get the following error:

```
#=== ERROR while installing core_kernel.113.00.00 =============================#
# opam-version         1.2.2
# os                   cygwin
# command              make
# path                 /home/crd/.opam/4.02.3/build/core_kernel.113.00.00
# compiler             4.02.3
# exit-code            2
# env-file             /home/crd/.opam/4.02.3/build/core_kernel.113.00.00/core_kernel-8856-5f1023.en
v
# stdout-file          /home/crd/.opam/4.02.3/build/core_kernel.113.00.00/core_kernel-8856-5f1023.ou
t
# stderr-file          /home/crd/.opam/4.02.3/build/core_kernel.113.00.00/core_kernel-8856-5f1023.er
r
### stdout ###
# [...]
# /home/crd/.opam/4.02.3/bin/ocamlfind ocamlc -g -c src/bigstring_marshal_stubs.c
# mv bigstring_marshal_stubs.o src/bigstring_marshal_stubs.o
# /home/crd/.opam/4.02.3/bin/ocamlfind ocamlc -g -c src/bigstring_stubs.c
# + /home/crd/.opam/4.02.3/bin/ocamlfind ocamlc -g -c src/bigstring_stubs.c
# src/bigstring_stubs.c:34:24: fatal error: sys/endian.h: No such file or directory
#  #include <sys/endian.h>
#                         ^
# compilation terminated.
# Command exited with code 2.
# Makefile:21: recipe for target 'build' failed
### stderr ###
# W: Cannot find source file matching module 'core_kernel' in library core_kernel
# E: Failure("Command ''/home/crd/.opam/4.02.3/bin/ocamlbuild' src/libcore_kernel_stubs.a src/dllcore_kernel_stubs.so src/core_kernel.cma src/core_kernel.cmxa src/core_kernel.a src/core_kernel.cmxs -use-ocamlfind -tag debug' terminated with error code 10")
# make: *** [build] Error 1
```

The following patch allows building:

```
--- src/bigstring_stubs.c.orig       2015-08-18 07:14:45.000000000 -0400
+++ src/bigstring_stubs.c       2016-01-15 21:01:17.687776600 -0500
@@ -29,6 +29,8 @@
 #define bswap_16 swap16
 #define bswap_32 swap32
 #define bswap_64 swap64
+#elif __CYGWIN__
+#include <endian.h>
 #else
 #include <sys/types.h>
 #include <sys/endian.h>
```

I haven't figured out what the correct way to apply this through opam is but I did manage to get core_kernel to install via opam through the hack of rerolling the tarball and changing checksum in the urls.txt file.

**Comment by diml** [comment]
Thanks, the fix will be part of the 113.24 release

**Comment by hhugo** [comment]
113.24 was released long ago
